home/import_ordered_content_sets.py

```diff
@@ -0,0 +1,253 @@
+import csv
+import io
+from dataclasses import dataclass
+from io import BytesIO
+from logging import getLogger
+from queue import Queue
+
+from django.core.files.base import File  # type: ignore
+from openpyxl import load_workbook
+
+from home.import_helpers import ImportException
+from home.models import ContentPage, OrderedContentSet
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class OrderedContentSetPage:
+    time: str
+    unit: str
+    before_or_after: str
+    page_slug: str
+    contact_field: str
+
+
+class OrderedContentSetImporter:
+    def __init__(
+        self,
+        file: File,
+        file_type: str,
+        progress_queue: Queue[int],
+    ):
+        """
+        Initialize an instance of OrderedContentSetImporter.
+
+        :param file: The file to be imported, as a django.core.files.base.File.
+        :param file_type: The type of the file, e.g. 'CSV' or 'XLSX'.
+        :param progress_queue: A queue.Queue to put progress information on.
+        """
+        self.file = file
+        self.filetype = file_type
+        self.progress_queue = progress_queue
+
+    def _get_or_init_ordered_content_set(
+        self, row: dict[str, str], set_name: str
+    ) -> OrderedContentSet:
+        """
+        Get or initialize an instance of OrderedContentSet from a row of a CSV file.
+
+        :param row: The row of the CSV file, as a dict.
+        :param set_name: The name of the ordered content set.
+        :return: An instance of OrderedContentSet.
+        """
+        ordered_set = OrderedContentSet.objects.filter(name=set_name).first()
+        if not ordered_set:
+            ordered_set = OrderedContentSet(name=set_name)
+
+        return ordered_set
+
+    def _add_profile_fields(
+        self, ordered_set: OrderedContentSet, row: dict[str, str]
+    ) -> None:
+        """
+        Add profile fields to an instance of OrderedContentSet from a row of a CSV file.
+
+        :param ordered_set: An instance of OrderedContentSet.
+        :param row: The row of the CSV file, as a dict.
+        """
+        ordered_set.profile_fields = []
+        for field in [f.strip() for f in (row["Profile Fields"] or "").split(",")]:
+            if field and field != "-":
+                [field_name, field_value] = field.split(":")
```

Comment:
Are we guaranteed to have exactly one `:`? If not, can we maybe raise an `ImportException` with a suitable message if we have the wrong number?

Reply:
This would be stuff like gender:male, so yes

Reply:
Added a ticket for enforcing uniqueness in names: https://praekelt.leankit.com/card/31512228558429

Reply:
> This would be stuff like gender:male, so yes

That would be a requirement and expectation, not a guarantee. Perhaps we could use some tests for `gender::male` and `gender-male` and such to make sure we're reporting content bugs instead of failing with no useful feedback to the user?